oauthproxy.go

```diff
@@ -115,6 +117,53 @@ func NewFileServer(path string, filesystemPath string) (proxy http.Handler) {
 	return http.StripPrefix(path, http.FileServer(http.Dir(filesystemPath)))
 }
 
+func CreateSecureProxy(opts *Options, validator func(string) bool) http.Handler {
+	bareproxy := NewOAuthProxy(opts, validator)
+	err := opts.Validate()
+	if err != nil {
+		log.Printf("%s", err)
+		os.Exit(1)
+	}
+
+	if len(opts.EmailDomains) != 0 && opts.AuthenticatedEmailsFile == "" {
+		if len(opts.EmailDomains) > 1 {
+			bareproxy.SignInMessage = fmt.Sprintf("Authenticate using one of the following domains: %v", strings.Join(opts.EmailDomains, ", "))
+		} else if opts.EmailDomains[0] != "*" {
+			bareproxy.SignInMessage = fmt.Sprintf("Authenticate using %v", opts.EmailDomains[0])
+		}
+	}
+
+	if opts.HtpasswdFile != "" {
+		log.Printf("using htpasswd file %s", opts.HtpasswdFile)
+		bareproxy.HtpasswdFile, err = NewHtpasswdFromFile(opts.HtpasswdFile)
+		bareproxy.DisplayHtpasswdForm = opts.DisplayHtpasswdForm
+		if err != nil {
+			log.Fatalf("FATAL: unable to open %s %s", opts.HtpasswdFile, err)
+		}
+	}
+
+	secureMiddleware := secure.New(secure.Options{
```

Comment:
No need to recreate a variable, then return it--we can just return the output of `secure.New`.

Reply:
Isn't this returning the `http.Handler` though? and not the output of `secure.New`?